OpenRA.Mods.Common/Traits/TransformOnCrush.cs

```diff
@@ -0,0 +1,57 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class TransformOnCrushInfo : ITraitInfo
+	{
+		[ActorReference, FieldLoader.Require] public readonly string IntoActor = null;
+
+		public readonly bool SkipMakeAnims = true;
+
+		public readonly HashSet<string> CrushClasses = new HashSet<string>();
+
+		public virtual object Create(ActorInitializer init) { return new TransformOnCrush(init, this); }
+	}
+
+	public class TransformOnCrush : INotifyCrushed
```

Comment:
The trait name suggests that the crushed actor will transform into something else, but it actually force-transforms the *crusher*.

Reply:
What do you suggest instead? Seems pretty clear to me.

Reply:
`TransformCrusherOnCrush` would be the simplest option.